File: projectforge-business/src/main/java/org/projectforge/sms/SMSSenderConfig.java
package org.projectforge.sms;

import org.apache.commons.lang3.StringUtils;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.stereotype.Controller;
import org.springframework.stereotype.Service;

import java.io.Serializable;
import java.util.Map;

@Configuration
public class SMSSenderConfig {

  public enum HttpMethodType {POST, GET}

  @Value("${projectforge.sms.httpMethod}")
  private HttpMethodType httpMethodType;
  @Value("${projectforge.sms.url}")
  private String url;
  @Value("#{${projectforge.sms.httpParameters}}")
  private Map<String, String> httpParams;
  @Value("${projectforge.sms.returnCodePattern.success}")
  private String smsReturnPatternSuccess;
  @Value("${projectforge.sms.returnCodePattern.numberError}")
  private String smsReturnPatternNumberError;
  @Value("${projectforge.sms.returnCodePattern.messageToLargeError}")
  private String smsReturnPatternMessageToLargeError;
  @Value("${projectforge.sms.returnCodePattern.messageError}")
  private String smsReturnPatternMessageError;
  @Value("${projectforge.sms.returnCodePattern.error}")
  private String smsReturnPatternError;
  @Value("${projectforge.sms.smsMaxMessageLength}")
  private int smsMaxMessageLength = 160;

  public String getSmsReturnPatternSuccess() {
    return smsReturnPatternSuccess;
  }

  public SMSSenderConfig setSmsReturnPatternSuccess(String smsReturnPatternSuccess) {
    this.smsReturnPatternSuccess = smsReturnPatternSuccess;
    return this;
  }

  public String getSmsReturnPatternError() {
    return smsReturnPatternError;
  }

  public SMSSenderConfig setSmsReturnPatternError(String smsReturnPatternError) {
    this.smsReturnPatternError = smsReturnPatternError;
    return this;
  }

  public String getSmsReturnPatternMessageError() {
    return smsReturnPatternMessageError;
  }

  public SMSSenderConfig setSmsReturnPatternMessageError(String smsReturnPatternMessageError) {
    this.smsReturnPatternMessageError = smsReturnPatternMessageError;
    return this;
  }

  public String getSmsReturnPatternMessageToLargeError() {
    return smsReturnPatternMessageToLargeError;
  }

  public SMSSenderConfig setSmsReturnPatternMessageToLargeError(String smsReturnPatternMessageToLargeError) {
    this.smsReturnPatternMessageToLargeError = smsReturnPatternMessageToLargeError;
    return this;
  }

  public String getSmsReturnPatternNumberError() {
    return smsReturnPatternNumberError;
  }

  public SMSSenderConfig setSmsReturnPatternNumberError(String smsReturnPatternNumberError) {
    this.smsReturnPatternNumberError = smsReturnPatternNumberError;
    return this;
  }

  public HttpMethodType getHttpMethodType() {
    return httpMethodType;
  }

  public SMSSenderConfig setHttpMethodType(HttpMethodType httpMethodType) {
    this.httpMethodType = httpMethodType;
    return this;
  }

  public SMSSenderConfig setHttpMethodType(String httpMethodType) {
    this.httpMethodType = StringUtils.equalsIgnoreCase("get", httpMethodType) ? HttpMethodType.GET : HttpMethodType.POST;
    return this;
  }

  public int getSmsMaxMessageLength() {
    return smsMaxMessageLength;
  }

  public SMSSenderConfig setSmsMaxMessageLength(int smsMaxMessageLength) {
    this.smsMaxMessageLength = smsMaxMessageLength;
    return this;
  }

  public String getUrl() {
    return url;
  }

  public SMSSenderConfig setUrl(String url) {
    this.url = url;
    return this;
  }

  public Map<String, String> getHttpParams() {
    return httpParams;
  }

  public SMSSenderConfig setHttpParams(Map<String, String> httpParams) {
    this.httpParams = httpParams;
    return this;
  }

  public boolean isSmsConfigured() {
    return StringUtils.isNotBlank(url);
  }
}

File: projectforge-business/src/main/kotlin/org/projectforge/sms/SmsSenderConfig.kt
/////////////////////////////////////////////////////////////////////////////
//
// Project ProjectForge Community Edition
//         www.projectforge.org
//
// Copyright (C) 2001-2019 Micromata GmbH, Germany (www.micromata.com)
//
// ProjectForge is dual-licensed.
//
// This community edition is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License as published
// by the Free Software Foundation; version 3 of the License.
//
// This community edition is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
// Public License for more details.
//
// You should have received a copy of the GNU General Public License along
// with this program; if not, see http://www.gnu.org/licenses/.
//
/////////////////////////////////////////////////////////////////////////////

package org.projectforge.sms

import org.apache.commons.lang3.StringUtils
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration


@Configuration
open class SmsSenderConfig {
    enum class HttpMethodType {
        POST, GET
    }

    @Value("\${projectforge.sms.httpMethod}")
    open var httpMethodType: HttpMethodType? = null

    @Value("\${projectforge.sms.url}")
    open var url: String? = null

    @Value("#{\${projectforge.sms.httpParameters}}")
    open var httpParams: Map<String, String>? = null

    @Value("\${projectforge.sms.returnCodePattern.success}")
    open var smsReturnPatternSuccess: String? = null

    @Value("\${projectforge.sms.returnCodePattern.numberError}")
    open var smsReturnPatternNumberError: String? = null

    @Value("\${projectforge.sms.returnCodePattern.messageToLargeError}")
    open var smsReturnPatternMessageToLargeError: String? = null

    @Value("\${projectforge.sms.returnCodePattern.messageError}")
    open var smsReturnPatternMessageError: String? = null

    @Value("\${projectforge.sms.returnCodePattern.error}")
    open var smsReturnPatternError: String? = null

    @Value("\${projectforge.sms.smsMaxMessageLength}")
    open var smsMaxMessageLength = 160

    fun isSmsConfigured(): Boolean {
        return StringUtils.isNotBlank(url)
    }

    fun setHttpMethodType(httpMethodType: String): SmsSenderConfig {
        this.httpMethodType = if (StringUtils.equalsIgnoreCase("get", httpMethodType)) HttpMethodType.GET else HttpMethodType.POST
        return this
    }
}
